src/Clock.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+// eslint-disable-next-line react/prefer-stateless-function
+export class Clock extends React.Component {
+  state = {
+    time: new Date(),
+  }
+
+  componentDidMount() {
+    this.interval = setInterval(
+      () => this.setState({ time: new Date() }),
+      1000,
+    );
+  }
+
+  componentDidUpdate(prevProps) {
+    const oldName = prevProps.clockName;
+    const curName = this.props.clockName;
+
+    if (oldName !== curName) {
+      console.log(curName);
+    }
+  }
```

Comment:
It's great that you have used componentDidUpdate, but for this task, you should only console.log the new name only when clicking the button (what you are actually doing in your 'clockNewName'). No need in this 'componentDidUpdate'